Makefile

```
@@ -151,6 +151,14 @@ build-python-lib:
 clean-python:
 	@rm -rf python/build
 
+zql/zql.go: zql/zql.peg
+	(cd zql ; make)
+
+peg: zql/zql.go FORCE
+	go run ./cmd/ast -repl
+
+FORCE:
+
```

```suggestion
peg:
	$(MAKE) -C zql
	go run ./cmd/ast -repl
```
`zql/zql.go: zql/zql.peg` and `peg: zql/zql.go` don't accurately reflect the dependencies.

Also, since this runs the ast command, I think ast would be a better name for this goal than peg.

---

The move to `-C` is good but I don't understand your comments here.  Your change will cause pigeon and pegjs to run every time, even when zql/zql.peg has't been modified.  I believe my rules are correct.  They say that zq/zql.go depends on zql/zql.peg and the peg rule says that if I want to run the ast command on the current version of zql/zql.peg, then it needs to be updated if it's stale, but otherwise it can run.  The output from the rules is very helpful because if you're editing along and it is run everytime, but you happen to be editing the wrong zql.peg, then the feedback from make on triggering the build is very helpful.  And when you just want to type `!make` and it re-runs pigeon and pegjs when it doesn't have to, well I find that to be an unnecessary delay in my workflow.  It might be that I've completely missed your point though, so let me know what you think of my comments here...  p.s. I like `make peg` better than `make ast` because it's meant to be used in an edit-compile-debug cycle and you are building the peg grammar, not the ast command.

---

zql.go depends on zql.peg, parser-support.go, and the installed versions of goimports and pigeon.

I understand your point about unnecessary delay. At the same time, I'm wary of hiding a surprise for others inside a makefile that doesn't reflect accurately reflect dependencies. Let's see if we can come up with something fast and accurate.

---

Replacing `$(MAKE) -C zql` with `$(MAKE) -C zql zql.go` trims that step from 4.3 seconds to 3.2 seconds for me, but that's still slow.

---

zql.go does not depend on parser-support.go in the build sense but you have a good point about the dependency on version info in zql/Makefile.  I just pushed another change that should address this.  What do you think now?

---

> zql.go does not depend on parser-support.go in the build sense

Right you are. I'd forgotten that peg.js includes only parser-support.js.